src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -313,7 +312,7 @@ def data_source_data_to_pandas_df(
     def df_map_method(pandas_df):
         return (
             pandas_df.applymap
-            if version.parse(pd.__version__) < version.parse("2.1.0")
+            if tuple(map(int, (str(pd.__version__).split(".")))) < (2, 1, 0)
```

Comment:
https://github.com/snowflakedb/snowflake/blob/caa286eb8f6b792ee01a64b72407602c4a8e0566/Tests/jenkins/PythonStoredProcedure/bump_regression_test_version.py#L223-L225

for handling alpha versions